### PR TITLE
Fix fork documentation

### PR DESCRIPTION
**Pull request by pavel-mukhanov**
## Overview
Fix `fork` documentation related to checkpoints.

---
### Definition of Done

- [ ] There are no TODOs left in the merged code
- [ ] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: components/merkledb/src/db.rs [hunk 1]
```diff
@@ -565,23 +564,15 @@ impl Snapshot for FlushedFork {
...
 impl Fork {
-    /// Finalizes all changes after the latest checkpoint.
-    ///
-    /// # Panics
-    ///
-    /// Panics if there is no active checkpoint, or the latest checkpoint
-    /// is already committed or rolled back.
+    /// Finalizes all changes that was made after previous execution of itself.
+    /// If no `flush` method was called before, finalizes all changes that was
+    /// made after creation of `Fork`. In other words creates checkpoint.
```
Comment (popzxc) [inline components/merkledb/src/db.rs]: I think the last sentence is redundant here.

Comment (Maria-Nosyk) [inline components/merkledb/src/db.rs]: /// Finalizes all changes that were made after previous execution of the `flush` method.
 /// If no `flush` method had been called before, finalizes all changes that were
/// made after creation of `Fork`.

File: components/merkledb/src/db.rs [hunk 2]
```diff
@@ -565,23 +564,15 @@ impl Snapshot for FlushedFork {
...
         working_patch.merge_into(&mut self.flushed.patch);
     }
 
-    /// Rolls back all changes after the latest checkpoint.
-    ///
-    /// # Panics
-    ///
-    /// Panics if there is no active checkpoint, or the latest checkpoint
-    /// is already committed or rolled back.
+    /// Rolls back all changes that was made after the latest execution of `flush`.
```
Comment (Maria-Nosyk) [inline components/merkledb/src/db.rs]: that were made after the latest execution of the `flush` method.

File: components/merkledb/src/db.rs [hunk 3]
```diff
@@ -565,23 +564,16 @@ impl Snapshot for FlushedFork {
...
         working_patch.merge_into(&mut self.flushed.patch);
     }
 
-    /// Rolls back all changes after the latest checkpoint.
-    ///
-    /// # Panics
-    ///
-    /// Panics if there is no active checkpoint, or the latest checkpoint
-    /// is already committed or rolled back.
+    /// Rolls back all changes that that were made after the latest execution
```
Comment (Maria-Nosyk) [inline components/merkledb/src/db.rs]: > changes that that were

extra "that"

File: components/merkledb/src/db.rs [hunk 4]
```diff
@@ -370,9 +370,9 @@ pub enum Change {
 /// need to consistently apply several sets of changes to the same data, the next fork should be
 /// created after the previous fork has been merged.
 ///
-/// `Fork` also supports checkpoints ([`checkpoint`], [`commit`] and
+/// `Fork` also supports checkpoints ([`flush`] and
 /// [`rollback`] methods), which allows rolling back some of the latest changes (e.g., after
-/// a runtime error).
+/// a runtime error). Checkpoint is created automatically after calling `flush` method.
```
Comment (Maria-Nosyk) [inline components/merkledb/src/db.rs]: calling the `flush` method.

**Pull request merged**